service/src/main/java/bio/terra/workspace/service/spendprofile/SpendProfileService.java

```diff
@@ -55,7 +76,14 @@ public SpendProfile authorizeLinking(
         "isAuthorized")) {
       throw SpendUnauthorizedException.linkUnauthorized(spendProfileId);
     }
-    SpendProfile spend = spendProfiles.get(spendProfileId);
+
+    SpendProfile spend;
+    if (bpmEnabled) {
```

Comment:
🤔 Good point. `LINK` is implied by a billing profile coming back from the service.

Reply:
That being said, there is no explicit check for `LINK` in BPM, it's implied since all owners and users have `LINK`.  If the SAM model changes that could open a hole; i.e., I'm a billing administrator and want to view the billing profiles but can not spend against them.

I think I agree that we want BPM to be making these checks rather than leaking them to external services, perhaps we can have BPM make an explicit LINK check and if the need arises for readonly profiles we can fashion a new endpoint.